openedx/features/edly/models.py

```diff
@@ -0,0 +1,30 @@
+from django.contrib.sites.models import Site
+from django.core.validators import RegexValidator
+from django.db import models
+from organizations.models import Organization
+
+ALPHANUMERIC_VALIDATOR = RegexValidator(r'^[0-9a-zA-Z]*$', 'Only alphanumeric characters are allowed.')
+
+
+class EdlyOrganization(models.Model):
+    name = models.CharField(max_length=255)
+    slug = models.CharField(max_length=100, unique=True, validators=[ALPHANUMERIC_VALIDATOR])
+
+    def __str__(self):
+        return u"{name} ({slug})".format(name=self.name, slug=self.slug)
+
+
+class EdlySubOrganization(models.Model):
+    edly_organization = models.ForeignKey(EdlyOrganization, null=False, blank=False)
+    edx_organization = models.OneToOneField(Organization)
```

Comment:
We need to make sure that there is no existing `edx_organization` with the `short_name` provided by client or us.

Reply:
this is already edx-organization app model.. and there is already a db_index=True in it but not unique constraint applied

Reply:
How about adding a unique together `edly_organization`, `edx_organization`?
One unique `edx_organization` (edly_sub_organization) should only be linked to one `edly_organization`

Same for `edx_organization`, `lms_site` and `studio_site`.
One `edx_organization` (edly_sub_organization) should only be linked to unique one `lms_site` and `studio_site`.

Reply:
Done

Reply:
Please add unique together constraints for sites as well.

Reply:
There is already a constraint on unique sites.
<img width="1641" alt="Screenshot 2020-05-04 at 6 00 08 PM" src="https://user-images.githubusercontent.com/17013371/80968293-2431b300-8e31-11ea-9b62-4940edc9832f.png">

Reply:
I am talking about uniqueness of sites with `edx_organization`

Reply:
sites have one-to-one relationship with edly_suborganization. and edly_suborganization has a one-to-one relationship with self-organization.

Reply:
Ok. good.

Reply:
Ok.